x-pack/plugins/maps/public/connected_components/right_side_controls/layer_control/layer_toc/toc_entry/toc_entry_actions_popover/toc_entry_actions_popover.tsx

```diff
@@ -34,10 +39,61 @@ export interface Props {
 
 interface State {
   isPopoverOpen: boolean;
+  supportsFeatureEditing: boolean;
+  editModeEnabled: boolean;
 }
 
 export class TOCEntryActionsPopover extends Component<Props, State> {
-  state: State = { isPopoverOpen: false };
+  state: State = { isPopoverOpen: false, supportsFeatureEditing: false, editModeEnabled: false };
+  private _isMounted = false;
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidUpdate() {
+    this._checkLayerEditable();
+  }
+
+  async _checkLayerEditable() {
+    if (!(this.props.layer instanceof VectorLayer)) {
+      return;
+    }
+    const supportsFeatureEditing = this.props.layer.supportsFeatureEditing();
+    const editModeEnabled = await this._getEditModeEnabled();
+    if (
+      !this._isMounted ||
+      (supportsFeatureEditing === this.state.supportsFeatureEditing &&
+        editModeEnabled === this.state.editModeEnabled)
+    ) {
+      return;
+    }
+    this.setState({ supportsFeatureEditing, editModeEnabled });
+  }
+
+  async _getEditModeEnabled(): Promise<boolean> {
+    const vectorLayer = this.props.layer as VectorLayer;
+    const layerSource = await this.props.layer.getSource();
+    if (!(layerSource instanceof ESSearchSource)) {
+      return false;
+    }
+    // @ts-ignore
```

Comment:
Can @ts-ignore be removed? What is the ts error?

Reply:
That is an odd one. Happy to remove the comment if you know the fix:
```
info x-pack/plugins/maps/public/connected_components/right_side_controls/layer_control/layer_toc/toc_entry/toc_entry_actions_popover/toc_entry_actions_popover.tsx:84:53 - error TS2339: Property 'scalingType' does not exist on type 'ESSearchSourceSyncMeta | ESGeoGridSourceSyncMeta | ESGeoLineSourceSyncMeta | ESTermSourceSyncMeta'.
        Property 'scalingType' does not exist on type 'ESGeoGridSourceSyncMeta'.

      84     const isClustered = layerSource?.getSyncMeta()?.scalingType === SCALING_TYPES.CLUSTERS;
```

But it does in fact exist on `ESSearchSourceSyncMeta` [here](http://github.com/aaronjcaldwell/kibana/blob/a9311dd83ae8f8b8df0583eab12ec18f7390bd90/x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.ts#L39-L39).

Reply:
Just case getSyncMeta() as ESSearchSourceSyncMeta

Reply:
Done, thanks. I'd stopped one step short of that casting the source as an `ESSearchSource`. Still not entirely sure why it doesn't work that way but I'll take the fix!